pkg/core/native/crypto.go

```diff
@@ -335,6 +397,285 @@ func (c *Crypto) bls12381Deserialize(_ *interop.Context, args []stackitem.Item)
 	return stackitem.NewInterop(*p)
 }
 
+func (c *Crypto) bls12381SerializeEth(_ *interop.Context, args []stackitem.Item) stackitem.Item {
+	val, ok := args[0].(*stackitem.Interop).Value().(blsPoint)
+	if !ok {
+		panic(errors.New("not a bls12381 point"))
+	}
+	return stackitem.NewByteArray(val.BytesEth())
+}
+
+func (c *Crypto) bls12381DeserializeEth(_ *interop.Context, args []stackitem.Item) stackitem.Item {
+	buf, err := args[0].TryBytes()
+	if err != nil {
+		panic(fmt.Errorf("invalid serialized bls12381 point: %w", err))
+	}
+	p := new(blsPoint)
+	err = p.FromBytesEth(buf)
+	if err != nil {
+		panic(err)
+	}
+	return stackitem.NewInterop(*p)
+}
+
+// bls12381SerializeList serializes a list of BLS12-381 points or (point, scalar) pairs.
+func (c *Crypto) bls12381SerializeList(_ *interop.Context, args []stackitem.Item) stackitem.Item {
+	items := args[0].Value().([]stackitem.Item)
+	if len(items) == 0 {
+		panic("at least one element is required")
+	}
+	var res []byte
+	if items[0].Type() == stackitem.InteropT {
+		for _, v := range items {
+			p, ok := v.(*stackitem.Interop).Value().(blsPoint)
+			if !ok {
+				panic("not a bls12381 point")
+			}
+			res = append(res, p.Bytes()...)
+		}
+		return stackitem.NewByteArray(res)
+	}
+	for _, si := range items {
+		if si.Type() != stackitem.ArrayT && si.Type() != stackitem.StructT {
+			panic("pair must be Array or Struct")
+		}
+		pair := si.Value().([]stackitem.Item)
+		if len(pair) != 2 {
+			panic("pair must contain point and scalar")
+		}
+		scalar, ok := pair[1].Value().(*big.Int)
+		if !ok {
+			panic("scalar must be bigint")
+		}
+		p, ok := pair[0].(*stackitem.Interop).Value().(blsPoint)
+		if !ok {
+			panic("not a bls12381 point")
+		}
+		res = append(res, p.Bytes()...)
+		scalarBytes := make([]byte, fr.Bytes)
+		bigint.ToPreallocatedBytes(scalar, scalarBytes)
+		res = append(res, scalarBytes...)
+	}
+	return stackitem.NewByteArray(res)
+}
+
+func (c *Crypto) bls12381SerializeEthList(_ *interop.Context, args []stackitem.Item) stackitem.Item {
```

Comment:
Every introduced method needs a comment explaining which type of conversion it's performing, like it's done for `bls12381SerializeList`.